src/textual/geometry.py

```diff
@@ -137,100 +153,121 @@ class Region(NamedTuple):
     height: int = 0
 
     @classmethod
-    def from_corners(cls, x1: int, y1: int, x2: int, y2: int) -> Region:
-        """Construct a Region form the top left and bottom right corners.
+    def from_corners(cls: type[R], x1: int, y1: int, x2: int, y2: int) -> R:
+        """Construct a Region from the top-left and bottom-right corners.
 
         Args:
-            x1 (int): Top left x
-            y1 (int): Top left y
-            x2 (int): Bottom right x
-            y2 (int): Bottom right y
+            x1 (int): Top-left x
+            y1 (int): Top-left y
+            x2 (int): Bottom-right x
+            y2 (int): Bottom-right y
 
         Returns:
             Region: A new region.
         """
         return cls(x1, y1, x2 - x1, y2 - y1)
 
     @classmethod
-    def from_origin(cls, origin: tuple[int, int], size: tuple[int, int]) -> Region:
-        """Create a region from origin and size.
+    def from_origin(cls: type[R], origin: tuple[int, int], size: tuple[int, int]) -> R:
+        """Create a new region from `origin` and `size`.
 
         Args:
-            origin (Point): Origin (top left point)
-            size (tuple[int, int]): Dimensions of region.
+            origin (tuple[int, int]): The origin (top-left point) of the new region.
+            size (tuple[int, int]): The dimensions of the new region.
 
         Returns:
-            Region: A region instance.
+            Region: A new `Region` instance.
         """
         x, y = origin
         width, height = size
         return cls(x, y, width, height)
 
     def __bool__(self) -> bool:
+        """Return True if the width and height of the region are both nonzero."""
         return bool(self.width and self.height)
 
     @property
     def x_extents(self) -> tuple[int, int]:
-        """Get the starting and ending x coord.
+        """Get the starting and ending x-coordinates of the region.
 
-        The end value is non inclusive.
+        The end value is non-inclusive.
 
         Returns:
-            tuple[int, int]: [description]
+            tuple[int, int]: A tuple describing the minimum and maximum x-coordinates.
         """
-        return (self.x, self.x + self.width)
+        x = self.x
```

Comment:
Why make this two lines? If it's faster, I'll need proof! :-)

Reply:
Timings appear to indicate a tiny speed increase, but barely noticeable (weirdly, it seems to be a bigger speed difference when done with fewer repeats -- but I'm guessing that's just noise). I'll revert the change -- sorry for wasting your time.
[geometry.Region.x_extents timings.txt](https://github.com/willmcgugan/textual/files/7243798/geometry.Region.x_extents.timings.txt)